### ISSUE TITLE
Cannot register services with get_it as an abstraction - 'asType' argument missing

### ISSUE THREAD

**Issue by adamzemmoura** [Stacked-Org/stacked#329]
I am trying to register a class as a LazySingleton with get_it and would like to register it as its interface rather than the implementation.  I see the super class of LazySingleton - DependencyRegistration - has asType and the documentation also gives the following example: 

> // abstracted class type support
    LazySingleton(classType: FirebaseAuthService, asType: AuthService),

Unless I am missing something, the asType argument is missing from the current LazySingleton implementation.  Actually it's missing from all 4 dependancy types.  

Is there some other way we're supposed to register services as an abstraction?

**Comment by FilledStacks** [comment]
Must have been merged out. Let me check.

**Comment by FilledStacks** [comment]
Can you check v2.0.3 please. It's still in all the annotations

**Comment by adamzemmoura** [comment]
Thanks for the quick reply.  I was on v2.0.2 which seems to be the latest version? I tried to upgrade to 2.0.3 but that version doesn't seem to exist?

Output from console:  `depends on stacked ^2.0.3 which doesn't match any versions, version solving failed.`

**Comment by adamzemmoura** [comment]
Seems like the last version with asType in the signature is v1.9.8

**Comment by FilledStacks** [comment]
Can you check in 2.0.3+1

I don't think I published 2.0.3. It's in the annotations file

```dart
/// The class to describe a service registration on the get_it locator
class DependencyRegistration {
  /// The type of the service to register
  final Type? classType;

  /// An abstracted class type of service to register
  final Type? asType;

  final Function? resolveUsing;

  const DependencyRegistration({this.classType, this.asType, this.resolveUsing});
}

/// Registers the type passed in as a singleton instance in the get_it locator
class Singleton extends DependencyRegistration {
  const Singleton({Type? classType, Type? asType, Function? resolveUsing})
      : super(
          classType: classType,
          asType: asType,
          resolveUsing: resolveUsing,
        );
}

/// Registers the type passed in as a LazySingleton instance in the get_it locator
class LazySingleton extends DependencyRegistration {
  const LazySingleton({Type? classType, Type? asType, Function? resolveUsing})
      : super(
          classType: classType,
          asType: asType,
          resolveUsing: resolveUsing,
        );
}

/// Registers the type passed in as a Factory in the get_it locator
class Factory extends DependencyRegistration {
  const Factory({Type? classType, Type? asType}) : super(classType: classType, asType: asType);
}

/// Registers the type passed in to be presolved using the function passed in
class Presolve extends DependencyRegistration {
  /// The static instance Future function to use for resolving the type registered
  final Future Function()? presolveUsing;

  const Presolve({Type? classType, Type? asType, this.presolveUsing})
      : super(classType: classType, asType: asType);
}

```

**Comment by adamzemmoura** [comment]
Working in v2.0.3+1 👍 Thanks again for your quick replies, for building this awesome package, along with the great content on YouTube. Keep up the great work! 🚀
